private-templates-service/client/src/AuthenticatedComponent.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { Redirect } from "react-router-dom";
+import { UserType } from "./App";
+import Welcome from "./components/Welcome/Welcome";
+
+interface State {
+}
+interface AuthProps {
+  isAuthenticated: boolean,
+  user?: UserType,
+  authButtonMethod: () => Promise<void>
+}
+
+export function requireAuthentication(Component: any) {
```

Comment:
This should be a generic, so that the component that uses it can pass its props too:

Something like:
`const  requireAuthentication = <P extends object>(Component: React.ComponentType<P>) => {

  return class AuthenticatedComponent extends React.Component<AuthProps & P> {`

Reply:
If you want me to explain this further let me know